### PR TITLE
[MO] - [system test] -> deployment resource not specifying namespace …

### PR DESCRIPTION
**Pull request by see-quick**
…when get

Signed-off-by: morsak <xorsak02@stud.fit.vutbr.cz>

### Type of change

- Bugfix

### Description

This PR fixes namespace handling in DeploymentResource, where in some edge cases the Kubernetes client could not get the resource in the correct namespace.

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

File: systemtest/src/main/java/io/strimzi/systemtest/resources/kubernetes/DeploymentResource.java [hunk 1]
```diff
@@ -20,7 +20,7 @@ public String getKind() {
     }
     @Override
     public Deployment get(String namespace, String name) {
-        return ResourceManager.kubeClient().namespace(namespace).getDeployment(name);
+        return ResourceManager.kubeClient().namespace(namespace).getDeployment(namespace, name);
```
Comment (Frawless) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/kubernetes/DeploymentResource.java:23]: Does it point to some problem in fabric8? Or why do we need to specify namespace twice?

Comment (see-quick) [reply systemtest/src/main/java/io/strimzi/systemtest/resources/kubernetes/DeploymentResource.java:23]: No, but in the case of parallel execution, there is a potential error to have a race condition. Because this piece of code is not synchronized, thus multiple threads could access it (with different namespaces) and it can change the `shared` state of the Kube client. Thus we need it explicitly specify `namespace` inside `getDeployment(...)` method.

Comment (see-quick) [reply systemtest/src/main/java/io/strimzi/systemtest/resources/kubernetes/DeploymentResource.java:23]: What I can do about it is remove `unnecessary` switching between namespaces from:
```java
return ResourceManager.kubeClient().namespace(namespace).getDeployment(namespace, name);
```
to
```java
return ResourceManager.kubeClient().getDeployment(namespace, name);
```

Comment (Frawless) [reply systemtest/src/main/java/io/strimzi/systemtest/resources/kubernetes/DeploymentResource.java:23]: Ye, that change would be nice.

Comment (see-quick) [reply systemtest/src/main/java/io/strimzi/systemtest/resources/kubernetes/DeploymentResource.java]: Changed )

**Review by Frawless — APPROVED** [review]
